File: epi_judge_python/int_as_array_increment.py
from test_framework import generic_test


def plus_one(A):
    idx = len(A) -1
    c=1
    while idx>-1:
        A[idx],c = (A[idx]+c,0) if A[idx]+c<=9 else (0,1)
        idx-=1
        if c==0:
            break
    if c==1:
        A.insert(0,1)
    return A


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main("int_as_array_increment.py",
                                       "int_as_array_increment.tsv", plus_one))

File: epi_judge_cpp/int_as_array_increment.cc
#include <vector>
#include "test_framework/generic_test.h"
using std::vector;
vector<int> PlusOne(vector<int> A)
{
  
  
  A.back()++;
  for (std::vector<int>::reverse_iterator it = std::rbegin(A); it !=std::rend(A)-1 &&  (*it)==10; it++)
  {
    (*it) =0 ;    
    (*(it+1))++;
  }

  if(A[0]==10){
    A[0] = 1;
    A.emplace_back(0);
  }


  

  return A;
}

int main(int argc, char *argv[])
{
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"A"};
  return GenericTestMain(args, "int_as_array_increment.cc",
                         "int_as_array_increment.tsv", &PlusOne,
                         DefaultComparator{}, param_names);
}
